server/server.go

```diff
@@ -2086,32 +2104,49 @@ func (s *Server) createClient(conn net.Conn, ws *websocket) *client {
 	now := time.Now()
 
 	c := &client{srv: s, nc: conn, opts: defaultOpts, mpay: maxPay, msubs: maxSubs, start: now, last: now, ws: ws}
+	if mqtt != nil {
+		c.mqtt = mqtt
+		// Set some of the options here since MQTT clients don't
+		// send a regular CONNECT (but have their own).
+		c.opts.Lang = "mqtt"
```

Comment:
Wonder if we should leave blank?

Reply:
No because some code check that Lang not empty means it is a "client" connection, as opposed to route, etc...

Reply:
I have removed it.